game-core/src/test/java/games/strategy/triplea/attachments/TriggerAttachmentTest.java

```diff
@@ -26,102 +29,226 @@
-    final Set<TriggerAttachment> satisfiedTriggers = Collections.singleton(triggerAttachment);
-
-    when(triggerAttachment.getProductionRule())
-        .thenReturn(Arrays.asList("frontier:rule1", "frontier:-rule2", "frontier:rule3"));
-    when(triggerAttachment.getName()).thenReturn("mockedTriggerAttachment");
-
-    final ProductionRuleList productionRuleList = gameData.getProductionRuleList();
-    productionRuleList.addProductionRule(new ProductionRule("rule1", gameData));
-    final ProductionRule productionRule2 = new ProductionRule("rule2", gameData);
-    productionRuleList.addProductionRule(productionRule2);
-    productionRuleList.addProductionRule(new ProductionRule("rule3", gameData));
-
-    final ProductionFrontierList productionFrontierList = gameData.getProductionFrontierList();
-    productionFrontierList
-        .addProductionFrontier(
-            new ProductionFrontier("frontier", gameData, Collections.singletonList(productionRule2)));
-
-    TriggerAttachment.triggerProductionFrontierEditChange(
-        satisfiedTriggers,
-        bridge,
-        "beforeOrAfter",
-        "stepName",
-        false, // useUses
-        false, // testUses
-        false, // testChance
-        false); // testWhen
-    final ArgumentCaptor<CompositeChange> argument = ArgumentCaptor.forClass(CompositeChange.class);
-    verify(bridge).addChange(argument.capture());
-    final CompositeChange change = argument.getValue();
-    assertFalse(change.isEmpty());
-    final ArgumentCaptor<String> ruleAddArgument = ArgumentCaptor.forClass(String.class);
-    verify(historyWriter, times(3)).startEvent(ruleAddArgument.capture());
-    final List<String> allValues = ruleAddArgument.getAllValues();
-    assertEquals(3, allValues.size());
-    assertTrue(allValues.stream().anyMatch(s -> s.contains("rule1") && s.contains("added")));
-    assertTrue(allValues.stream().anyMatch(s -> s.contains("rule2") && s.contains("removed")));
-    assertTrue(allValues.stream().anyMatch(s -> s.contains("rule3") && s.contains("added")));
-    assertTrue(allValues.stream().allMatch(s -> s.contains("frontier")));
+  void testGetClearFirstNewValue() {
+    {
+      final Tuple<Boolean, String> r = TriggerAttachment.getClearFirstNewValue("-clear-");
+      assertTrue(r.getFirst());
+      assertTrue(r.getSecond().isEmpty());
+    }
+
+    {
+      final Tuple<Boolean, String> r = TriggerAttachment.getClearFirstNewValue("-reset-");
+      assertTrue(r.getFirst());
+      assertTrue(r.getSecond().isEmpty());
+    }
+
+    {
+      final Tuple<Boolean, String> r = TriggerAttachment.getClearFirstNewValue("-clear-4:conscript");
+      assertTrue(r.getFirst());
+      assertEquals(r.getSecond(), "4:conscript");
+    }
+
+    {
+      final Tuple<Boolean, String> r = TriggerAttachment.getClearFirstNewValue("clearValueWithoutDash");
+      assertFalse(r.getFirst());
+      assertEquals(r.getSecond(), "clearValueWithoutDash");
+    }
```

Comment:
Using blocks is not really super pretty.
There's nothing wrong with creating a dedicated `@Test` method for every single one of those test cases.